File: SCT2/tags/ci-build/jenkins-YAKINDU_SCT2_CI-1459/test-plugins/org.yakindu.sct.model.sexec.interpreter.test/src-gen/org/yakindu/sct/model/sexec/interpreter/test/StringExpressionsTest.java
/**
 * Copyright (c) 2012 committers of YAKINDU and others.
 * All rights reserved. This program and the accompanying materials
 * are made available under the terms of the Eclipse Public License v1.0
 * which accompanies this distribution, and is available at
 * http://www.eclipse.org/legal/epl-v10.html
 *
 * Contributors:
 *     committers of YAKINDU - initial API and implementation
 */
package org.yakindu.sct.model.sexec.interpreter.test;
import org.eclipse.xtext.junit4.InjectWith;
import org.eclipse.xtext.junit4.XtextRunner;
import org.junit.Test;
import org.junit.runner.RunWith;
import org.yakindu.sct.model.sexec.interpreter.test.util.AbstractExecutionFlowTest;
import org.yakindu.sct.model.sexec.interpreter.test.util.SExecInjectionProvider;
import com.google.inject.Inject;
import org.junit.Before;
import org.yakindu.sct.model.sexec.interpreter.IExecutionFlowInterpreter;
import org.yakindu.sct.model.sexec.ExecutionFlow;
import util.TestModels;
import static junit.framework.Assert.*;
/**
 *  Unit TestCase for StringExpressions
 */
@SuppressWarnings("all")
@RunWith(XtextRunner.class)
@InjectWith(SExecInjectionProvider.class)
public class StringExpressionsTest extends AbstractExecutionFlowTest {

	@Inject
	private TestModels models;

	@Before
	public void setup() throws Exception {
		ExecutionFlow flow = models
				.loadExecutionFlowFromResource("StringExpressions.sct");
		initInterpreter(flow);
	}
	@Test
	public void StringExpressionsTest() throws Exception {
		interpreter.enter();
		assertTrue(isActive("StateA"));
		assertTrue(getString("myString").equals("hello"));
		assertTrue(getString("myString2").equals("world"));
		raiseEvent("e1");
		interpreter.runCycle();
		assertTrue(isActive("StateB"));
		assertTrue(getBoolean("equals") == false);
		assertTrue(getBoolean("notEqual") == true);
	}
}

File: SCT2/tags/ci-build/jenkins-YAKINDU_SCT2_CI-1983/test-plugins/org.yakindu.sct.generator.cpp.test/gtests/StringExpressionsTest/StringExpressionsTest.cc
/**
* Copyright (c) 2013 committers of YAKINDU and others.
* All rights reserved. This program and the accompanying materials
* are made available under the terms of the Eclipse Public License v1.0
* which accompanies this distribution, and is available at
* http://www.eclipse.org/legal/epl-v10.html
*
* Contributors:
*     committers of YAKINDU - initial API and implementation
*/
#include <string>
#include "gtest/gtest.h"
#include "StringExpressions.h"

TEST(StatemachineTest, StringExpressionsTest) {
	StringExpressions* statechart = new StringExpressions();
	statechart->init();
	EXPECT_TRUE(statechart->getSCInterface()->get_quotedString().compare("\"x\"") == 0);
	statechart->enter();
	EXPECT_TRUE(statechart->isActive(StringExpressions::StringExpressions_main_region_StateA));
	EXPECT_TRUE(statechart->getSCInterface()->get_myString().compare("hello") == 0);
	EXPECT_TRUE(statechart->getSCInterface()->get_myString2().compare("world") == 0);
	EXPECT_TRUE(statechart->getSCInterface()->get_quotedString().compare("'y'") == 0);
	statechart->raise_e1();
	statechart->runCycle();
	EXPECT_TRUE(statechart->isActive(StringExpressions::StringExpressions_main_region_StateB));
	EXPECT_TRUE(statechart->getSCInterface()->get_equals()== false);
	EXPECT_TRUE(statechart->getSCInterface()->get_notEqual()== true);
	EXPECT_TRUE(statechart->getSCInterface()->get_quotedString().compare("\"z\"") == 0);
	delete statechart;
}
